Clean_Data.R

```diff
@@ -0,0 +1,128 @@
+
+# create variable for social orgin
+unique(clean.2010$ocup1anc)
+clean.2010 <- clean.2010 %>%
+  mutate(parent_occ = ifelse(ocup1anc == "Professional, intellectual and scientist", 10, 
+                      ifelse(ocup1anc == "Director (manager, head of the department, supervisor)", 9,
+                      ifelse(ocup1anc == "Technician or mid-level professional", 8, 
+                      ifelse(ocup1anc == "Specialized worker", 3, 
+                      ifelse(ocup1anc == "Public official", 7, 
+                      ifelse(ocup1anc == "Office worker (secretary, offiice equipment operator, cashier, etc.)", 7,
+                      ifelse(ocup1anc == "Merchant (street vendor, owner of commercial establishment or market stand, etc.)", 4,
+                      ifelse(ocup1anc == "Warehous or market salesperson", 6,
+                      ifelse(ocup1anc == "Employed, outside an office, in the service sector (hotel or restaurant worker, taxi drivier, etc.)", 4,
+                      ifelse(ocup1anc == "Farm laborer, farmer, o agriculture and livestock producer agropecuario, and fisherman (owner of land)", 1,
+                      ifelse(ocup1anc == "Agricultural worker (works on land for others)", 1, 
+                      ifelse(ocup1anc == "Artisan, craftsperson", 5,
+                      ifelse(ocup1anc == "Domestic service", 1,
+                      ifelse(ocup1anc == "Laborer", 2, 
+                      ifelse(ocup1anc == "Member of the armed forces or protection and security services (the police, fireman, watchman, etc.)", 5, NA)))))))))))))))) 
+
+clean.2010$parent_occ <- factor(clean.2010$parent_occ)
+
+#check for 39,238 respondents in 23 of the 26 countries, as stated in paper
+write.dta(clean.2010, "clean2010data.dta")
+
+#output a dataset for graphs
+#get counts for colorr by country in temporary dataset
+test <- xtabs(formula = ~colorr + pais , data = clean.2010, exclude = "97 Colud not be classified")
+test <- as.data.frame(test)
+test$colorr <- as.numeric(test$colorr)
+
+#write a loop to organize recode
+#the loop that crashes freqs, first for low end values of colorr, then for high end to flag categories to collapse
+for(country in unique(test$pais)){
+  minInd = min(which(test$pais == country)) - 1
+  for (colorr in 1:6) { 
+    if (test[colorr +minInd, "Freq"] <= 30) {
+      test[colorr +minInd + 1, "Freq"] <- test[colorr + minInd, "Freq"] + test[colorr + minInd + 1, "Freq"]
+      test[colorr +minInd, "Freq"] <- 0 
+    }
+  }
+}
+
+for(country in unique(test$pais)){
+  minInd = min(which(test$pais == country)) - 1
+  for (colorr in 11:6) { 
+    if (test[colorr + minInd, "Freq"] <= 30) {
+        test[colorr + minInd - 1, "Freq"] <- test[colorr + minInd, "Freq"] + test[colorr + minInd - 1, "Freq"]
+        test[colorr + minInd, "Freq"] <- 0 
+    }
+  }
+}
+
```

Comment:
I'm glad that you got these working.  At the beginning of the loops maybe you could have some more documentation